pkg/lib/utils.jsx

```diff
@@ -31,3 +33,39 @@ export function fmt_to_fragments(fmt) {
 
     return React.createElement.apply(null, [React.Fragment, { }].concat(fmt.split(/(\$[0-9]+)/g).map(replace)));
 }
+
+function try_fields(dict, fields, def) {
+    for (let i = 0; i < fields.length; i++)
+        if (fields[i] && dict[fields[i]])
+            return dict[fields[i]];
+    return def;
+}
+
+/**
+ * Get an entry from a manifest's ".config[config_name]" field.
+ *
+ * This can either be a direct value, e.g.
+ *
+ *   "config": { "color": "yellow" }
+ *
+ * Or an object indexed by any value in "matches". Commonly these are fields
+ * from os-release(5) like PLATFORM_ID or ID; e.g.
+ *
+ *   "config": {
+ *      "fedora": { "color": "blue" },
+ *      "platform:el9": { "color": "red" }
+ *  }
+ */
+export function get_manifest_config_matchlist(manifest_name, config_name, default_value, matches) {
```

Comment:
c-machines can probably use this as well.

Reply:
Indeed, thanks for pointing out -- that's pretty much exactly what https://github.com/cockpit-project/cockpit-machines/blob/main/src/config.js does. I'll replace it after this lands (and a lib update)